src/controllers/mainController.js

```diff
@@ -2,16 +2,22 @@ const { mainService } = require("../services");
 const { catchAsync } = require("../middlewares");
 
 const getSearchResult = catchAsync(async (req, res, next) => {
-  const { query } = req.query;
+  const {
+    query: { query, filter, price, location, category },
+  } = req;
```

Comment:
가독성 면에서 더 나은 코드가 될 것 같습니다.
```suggestion
  const { query, filter, price, location, category } = req.query;
```